### PR TITLE
Major refactors to rendering pipeline and debug rendering

### PR DESCRIPTION
**Pull request by jake-pauls**
### Description
This PR makes multiple changes to how rendering works across the engine.

Firstly, it adds the `RenderBuddy` class that provides an API to access renderer methods. In this class are also the `DrawDebugLine` and `DrawDebugBox` methods which can be used to perform debug rendering on shapes (such as raycasts or bounding boxes). 

This PR also abstracts all rendering - for both models and lines into a `RenderingSystem` in an effort to begin cleaning up the update loop in `Application.cpp`. 🌵 

Finally, a `MathUtils` file was added for basic math operations (was required for experimenting with debug spheres - but will be useful in scripts later as well). 😎 

### Trello Ticket
https://trello.com/c/gEgURm96

### Additional Info
At the moment, calling a debug rendering method should always be called **outside** an update loop.

**Why?** - Our engine isn't mature enough for abstractions to be written for timers that will automatically kill the entities required for debug rendering or detect if the same entity is being added multiple times. Once a `Timer` utility is made we could easily add a `duration` parameter to these debug lines in order to kill them/add them in the ECS properly.

### CODE REVIEW

**Comment by jake-pauls** [comment]
[Add the ability to render debug lines/boxes in the engine](https://trello.com/c/gEgURm96/51-add-the-ability-to-render-debug-lines-boxes-in-the-engine)

**Comment by mbajaman** [comment]
@jake-pauls Question, so for the debug rendering methods, for example the DrawDebugLine, it takes vec3 from and vec3 to for the line itself.
Where would it get that info from?
Would that need to be taken from the rendered object itself and passed in?

**Review by mbajaman — APPROVED** [review]
Looks cleaaaan 🧼

**Comment by jake-pauls** [comment]
> @jake-pauls Question, so for the debug rendering methods, for example the DrawDebugLine, it takes vec3 from and vec3 to for the line itself. Where would it get that info from? Would that need to be taken from the rendered object itself and passed in?

Yep! This was more or less to primarily introduce `DrawDebugBox` as rendering bounding boxes might useful for visualizing collisions later. This also introduced the `DrawDebugLine` which might be useful if we end up needing to do raycasts for projectiles.

**tldr:** Say you had a projectile's start and end point (as `vec3`) you could draw the path it'll go on by passing it into the `to` and `from` arguments. You could also create these by constructing the vector inline (examples below).

```
// Draws a red debug line starting at (0, 0, -1) and ending at (0, 0, 1)
rad::RenderBuddy::DrawDebugLine(glm::Vec3(0.0f, 0.0f, -1.0f), glm::vec3(0.0f, 0.0f, 1.0f), glm::vec3(1.0f, 0.0f, 0.0f));

// Draws a green debug box with a minimum point (-0.5, -0.5, -0.5) and a maximum point (0.5, 0.5, 0.5)
rad::RenderBuddy::DrawDebugBox(glm::vec3(-0.5f, -0.5f, -0.5f), glm::vec3(0.5f, 0.5f, 0.5f), glm::vec3(0.0f, 1.0f, 0.0f));
```

**Review by kiwon132445 — APPROVED** [review]

**Review by Kotailri — APPROVED** [review]
